lib/src/iterable_extensions.dart

```diff
@@ -0,0 +1,725 @@
+// Copyright (c) 2020, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:collection/src/utils.dart';
+
+import 'algorithms.dart';
+
+/// Extensions that apply to all iterables.
+///
+/// These extensions provide direct access to some of the
+/// algorithms expose by this package,
+/// as well as some generally useful convenience methods.
+///
+/// More specialized extension methods that only apply to
+/// iterables with specific element types include those of
+/// [IterableComparableExtension] and [IterableNullableExtension].
+extension IterableExtension<T> on Iterable<T> {
+  /// The elements that do not satisfy [test].
+  Iterable<T> whereNot(bool Function(T element) test) =>
+      where((element) => !test(element));
+
+  /// Creates a sorted list of the elements of the iterable.
+  ///
+  /// The elements are ordered by the [compare] [Comparator].
+  List<T> sorted(Comparator<T> compare) => [...this]..sort(compare);
```

Comment:
Should we make this an optional argument to match `List.sort`?

Reply:
I forgot I had already asked this and we resolved the conversation. Sorry about that!

https://github.com/dart-lang/collection/pull/135#discussion_r425658615